sdc/tests/test_series.py

```diff
@@ -6175,6 +6133,596 @@ def test_impl(S):
         hpat_res = hpat_func(S)
         pd.testing.assert_series_equal(hpat_res, pandas_res)
 
+    @skip_sdc_jit("StringArray reflection was not implemented in old-pipeline")
+    @skip_numba_jit("TODO: support StringArray reflection")
+    def test_series_setitem_str_reflection(self):
+        """ Verifies that changes made to string Series passed into a jitted function
+            are propagated back to the native python object.
+        """
+        def test_impl(S, idx, val):
+            S[idx] = val
+            return S
+        hpat_func = self.jit(test_impl)
+
+        S = pd.Series(['cat', '', 'bbb', '', 'a', None, 'a', '', None, 'b'])
+        idx, value = 0, 'dog'
+        result = hpat_func(S, idx, value)
+        pd.testing.assert_series_equal(result, S)
+
+    @skip_numba_jit('Requires StringArray support of operator.eq')
+    def test_series_setitem_idx_str_scalar(self):
+        """ Verifies Series.setitem for scalar string idx operand and integer Series with index of matching dtype"""
+        def test_impl(A, i, value):
+            A[i] = value
+        hpat_func = self.jit(test_impl)
+
+        idx = 'a'
+        S = pd.Series(np.arange(5), index=['a', 'a', 'c', 'd', 'a'])
+        values_to_test = [-100,
+                          np.array([5, 55, 555]),
+                          pd.Series([5, 55, 555])]
+
+        for value in values_to_test:
+            with self.subTest(value=value):
+                S1 = S.copy(deep=True)
+                S2 = S.copy(deep=True)
+                hpat_func(S1, idx, value)
+                test_impl(S2, idx, value)
+                pd.testing.assert_series_equal(S1, S2)
+
+    def test_series_setitem_idx_int_scalar1(self):
+        """ Verifies Series.setitem for scalar integer idx operand and integer Series with index of matching dtype"""
+        def test_impl(A, i, value):
+            A[i] = value
+        hpat_func = self.jit(test_impl)
+
+        idx = 8
+        S = pd.Series(np.arange(5), index=[8, 8, 9, 8, 5])
+        values_to_test = [-100,
+                          np.array([5, 55, 555]),
+                          pd.Series([5, 55, 555])]
+
+        for value in values_to_test:
+            with self.subTest(value=value):
+                S1 = S.copy(deep=True)
+                S2 = S.copy(deep=True)
+                hpat_func(S1, idx, value)
+                test_impl(S2, idx, value)
+                pd.testing.assert_series_equal(S1, S2)
+
+    def test_series_setitem_idx_int_scalar2(self):
+        """ Verifies Series.setitem for scalar integer idx operand and integer Series with
+            index of non matching dtype (i.e. set along positions, not index)"""
+        def test_impl(A, i, value):
+            A[i] = value
+        hpat_func = self.jit(test_impl)
+
+        n = 11
+        np.random.seed(0)
```

Comment:
Look like you don't need to set up seed in this test.

Reply:
Right, thanks! I removed it here and in other places where there are no calls to np.random.